File: Fiona/my_robot_pack2/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"
using namespace std;

class MyFirstNode : public rclcpp::Node
{
public:
    MyFirstNode() : Node("first_node")
    {
        timer_ = this->create_wall_timer(
            chrono::seconds(1),
            bind(&MyFirstNode::timer_callback, this));
    }

private:
    void timer_callback()
    {
        RCLCPP_INFO(this->get_logger(), "Timer callback executed!");
    }

    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(make_shared<MyFirstNode>());
    rclcpp::shutdown();
    return 0;
}

File: Fiona/src/my_robot_package/my_robot_package/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyFirstNode(Node):
    def __init__(self):
        super().__init__('first_node')  # Initialize the node with a name
        self.create_timer(1.0, self.time_callback)  # Timer calls time_callback every second

    def time_callback(self):
        self.get_logger().info('Timer callback executed!')

def main(args=None):
    rclpy.init(args=args)  # Initialize the ROS 2 Python client library

    node = MyFirstNode()  # Instantiate your node class

    rclpy.spin(node)  # Keep the node alive

    rclpy.shutdown()

if __name__ == '__main__':
    main()
